test/sdk-execution-tests/sdk_execution_tests.py

```diff
@@ -0,0 +1,128 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import asyncio
+import dataclasses
+import functools
+import os
+import sys
+from typing import Any, Dict, List, Tuple
+
+from kfp import client
+from kfp import dsl  # noqa
+import kfp_server_api
+import pytest
+import yaml
+
+KFP_ENDPOINT = os.environ['KFP_ENDPOINT']
+TIMEOUT_SECONDS = os.environ['TIMEOUT_SECONDS']
+CURRENT_DIR = os.path.abspath(os.path.dirname(__file__))
+PROJECT_ROOT = os.path.abspath(
+    os.path.join(CURRENT_DIR, *([os.path.pardir] * 2)))
+CONFIG_PATH = os.path.join(PROJECT_ROOT, 'sdk', 'python', 'test_data',
+                           'test_data_config.yaml')
+DEFAULT_PIPELINE_FUNC_NAME = 'my_pipeline'
+
+kfp_client = client.Client(host=KFP_ENDPOINT)
+
+
+@dataclasses.dataclass
+class TestCase:
+    name: str
+    module_path: str
+    yaml_path: str
+    function_name: str
+    arguments: Dict[str, Any]
+
+
+def create_test_case_parameters() -> List[TestCase]:
+    parameters: List[TestCase] = []
+    with open(CONFIG_PATH) as f:
+        config = yaml.safe_load(f)
+    for name, test_group in config.items():
+        test_data_dir = os.path.join(PROJECT_ROOT, test_group['test_data_dir'])
+
+        parameters.extend(
+            TestCase(
+                name=name + '-' + test_case['module'],
+                module_path=os.path.join(test_data_dir,
+                                         f'{test_case["module"]}.py'),
+                yaml_path=os.path.join(test_data_dir,
+                                       f'{test_case["module"]}.yaml'),
+                function_name='my_pipeline' if name ==
+                'pipelines' else test_case['module'],
+                arguments=test_case.get('arguments'))
+            for test_case in test_group['test_cases']
+            if test_case['execute'])
+
+    return parameters
+
+
+def wait(run_result: client.client.RunPipelineResult) -> kfp_server_api.ApiRun:
+    return kfp_client.wait_for_run_completion(
+        run_id=run_result.run_id, timeout=int(TIMEOUT_SECONDS))
+
+
+def import_obj_from_file(python_path: str, obj_name: str) -> Any:
+    sys.path.insert(0, os.path.dirname(python_path))
+    module_name = os.path.splitext(os.path.split(python_path)[1])[0]
+    module = __import__(module_name, fromlist=[obj_name])
+    if not hasattr(module, obj_name):
+        raise ValueError(
+            f'Object "{obj_name}" not found in module {python_path}.')
+    return getattr(module, obj_name)
+
+
+def run(test_case: TestCase) -> Tuple[str, client.client.RunPipelineResult]:
+    full_path = os.path.join(PROJECT_ROOT, test_case.module_path)
+    pipeline_func = import_obj_from_file(full_path, test_case.function_name)
+    run_result = kfp_client.create_run_from_pipeline_func(
```

Comment:
What do you think about create run from precompiled YAML so that this execution test has no overlapping with compilation test? Say if we had an error in compilation logic, we won't say the same error across different tests.

Reply:
I like that -- will implement.

Reply:
On second thought, one shortcoming is that using the precompiled YAML doesn't allow us to mock the `kfp_package_path` (without some more involved string replacement) to make sure the PR branch's version of KFP is used. This is a big disadvantage, IMO, as these tests will be particularly useful for ensuring the lightweight component executor does not break.

Also, if the current branch were able to compile the pipeline to a precompiled snapshot, then we shouldn't have any duplicate errors here. And un-updated snapshots (YAML that is out of sync with the Python pipeline definition that generates it) should only throw errors in the read_write tests, not these execution tests.

I'm in favor of keeping as is. WDYT?

Reply:
>  one shortcoming is that using the precompiled YAML doesn't allow us to mock the kfp_package_path 

Okay, that's enough reason to recompile then. Thanks!